python/ray/autoscaler/_private/commands.py

```diff
@@ -691,6 +695,14 @@ def get_or_create_head_node(config: Dict[str, Any],
             setup_commands = config["head_setup_commands"]
             ray_start_commands = config["head_start_ray_commands"]
 
+        # If restarting Ray on a manually-managed on-prem cluster,
+        # we need to sync local and head representations of cluster state.
+        # If we're not restarting the cluster (empty ray start cmds), don't
+        # sync to avoid breaking on-prem cluster autoscaler state.
```

Comment:
Why is this related to ray start commands though?

Reply:
ray start commands are non-empty if and only if the ray cluster is being restarted.
I'll make this clearer with variable naming.

Reply:
It now reads more like English:
`if restarting_ray and is_local_manual_node_provider`

Reply:
The ray up machine thinks that worker nodes are terminated because it never creates worker nodes. It know that the head node is non-terminated. The only time it is acceptable to sync up that state is when ray is being started or restarted.